src/core/io/src/4C_io_input_spec_builders.hpp

```diff
@@ -2313,6 +2378,53 @@ Core::IO::InputSpec Core::IO::InputSpecBuilders::selection(
       common_data);
 }
 
+enum class InputFieldType : std::uint8_t
+{
+  constant,
+  from_file,
+};
```

Comment:
This currently lives at "global scope" (not really, it is inside `FourC` but we should not place it there either). If it is only used internally, put it into `Core::IO::Internal` to hide it from users. Same for the function below